### PR TITLE
[Analytics] Add GA script for Staging environment and general cleanup

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This pull request implements Google Analytics for our Staging environment. It also cleans up the way this code is included by -

- Moves the GA JS snippets into JS files named by environment
- Centralizes logic for importing GA code into `includes/google_analytics.liquid`
- Moves analytics code out of footer and into the header, so that we can collect analytics sooner. 
- Includes the snippets directly into the page to avoid the HTTP request

Ticket - https://github.com/department-of-veterans-affairs/vets.gov-team/issues/19247

## Testing done
- [x] Tested localhost
- [x] Tested vagovdev
- [x] Tested vagovstaging
- [x] Tested vagovprod

### buildtype - localhost
I don't think this is used, but it's the same as dev.va.gov.

image

### buildtype - vagovdev
image

### buildtype - vagovstaging
image

### buildtype - vagovprod
image
image

## Acceptance criteria
- [x] GA is loaded for all environments

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/includes/footer.html [hunk 1]
```diff
@@ -3,47 +3,10 @@
   <div id="footerNav"></div> <!-- let's try this React thing -->
 </footer>
 
-<!-- Begin generic overlay modal -->
-{% if modal %}
-  {% include "src/site/includes/modal.html" %}
-{% endif %}
```
Comment (ncksllvn) [inline src/site/includes/footer.html]: `modal.html` doesn't even exist anymore.

File: src/site/includes/footer.html [hunk 2]
```diff
@@ -3,47 +3,10 @@
...
-    if ( self !== top ) {
-      recordEvent({ event: 'iframed', embeddedIn: document.referrer});
-    }
-  </script>
-  <noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-WFJWBD"
-  height="0" width="0" class="csp-inline-patch-footer"></iframe></noscript>
-  <script async src="/js/google-analytics.js"></script>
-  <!-- End Google Tag Manager -->
-{% endif %}
-<!-- htmllint attr-bans="$previous" -->
```
Comment (ncksllvn) [inline src/site/includes/footer.html]: I moved the analytics code into the header, as recommended by the analytics team.

File: src/site/includes/footer.html [hunk 3]
```diff
@@ -3,47 +3,10 @@
...
 -->
-<script type="text/javascript">
-  function footerNavClick(elem) {
-    var eventLabel = 'nav-' + elem.parentElement.parentElement.id;
-    recordEvent({
-      event: eventLabel,
-      navlabel: elem.innerText
-    });
-  }
-</script>
```
Comment (ncksllvn) [inline src/site/includes/footer.html]: This is no longer used, since the footer was moved to React.

File: src/site/includes/header.html [hunk 4]
```diff
@@ -23,6 +23,20 @@
 <meta name="insight-app-sec-validation" content="ec59f99e-d869-4e3f-82d2-a61d32c65776">
 {% endif %}
 
+{% assign analytics = buildtype | default: 'vagovprod' %}
+{% if analytics == 'localhost' %}{% assign analytics = 'vagovdev' %}{% endif %}
+{% assign analytics = analytics | prepend: "src/site/assets/google-analytics/" %}
+{% assign analyticsScript = analytics | append: ".js" %}
+{% assign analyticsIframe = analytics | append: ".html" %}
```
Comment (ncksllvn) [inline src/site/includes/header.html]: We create paths to the GA files in `src/site/assets/google-analytics`, then just include them in the page.

File: src/site/assets/google-analytics/vagovdev.html [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-WFJWBD&gtm_auth=inC4EKQce9vlWpRVcowiyQ&gtm_preview=env-423&gtm_cookies_win=x"
+  height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
```
Comment (ncksllvn) [inline src/site/assets/google-analytics/vagovdev.html]: I don't know if `assets` is the perfect spot for this, but I want it close to the JS snippet, and I can't put the JS snippet anywhere else or else the linter will flip.

File: src/site/includes/header.html [hunk 6]
```diff
@@ -18,10 +18,9 @@
   {% include "src/site/includes/metatags.liquid" %}
 {% endif %}
 
-{% if buildtype == 'vagovstaging' %}
-<!-- Verification of site-ownership to the Insight App Sec Scanner -->
-<meta name="insight-app-sec-validation" content="ec59f99e-d869-4e3f-82d2-a61d32c65776">
-{% endif %}
+<script src="/js/record-event.js"></script>
+<script src="/js/settings.js"></script>
+{% include 'src/site/includes/google-analytics' %}
```
Comment (ncksllvn) [inline src/site/includes/header.html]: `record-event` has to be before `google-analytics` in order for `recordEvent` to be defined in the subsequent capture to the `buildtype`. So, I moved all of these JS files up a bit for faster tracking.

File: src/site/includes/header.html [hunk 7]
```diff
@@ -18,10 +18,9 @@
   {% include "src/site/includes/metatags.liquid" %}
 {% endif %}
 
-{% if buildtype == 'vagovstaging' %}
-<!-- Verification of site-ownership to the Insight App Sec Scanner -->
-<meta name="insight-app-sec-validation" content="ec59f99e-d869-4e3f-82d2-a61d32c65776">
-{% endif %}
```
Comment (ncksllvn) [inline src/site/includes/header.html]: This was used during WBC but isn't used anymore.

**Comment by ncksllvn** [comment]
Okay, this is ready for review!

**Review by erikphansen — APPROVED** [review]
lgtm!

**Pull request closed without merging**